adr/context.py

```diff
@@ -0,0 +1,287 @@
+import ast
+import inspect
+from jsone.interpreter import ExpressionEvaluator
+from jsone.prattparser import prefix
+
+COMMON_CONTEXTS = [
+    [['-B', '--branch'],
+     {'default': 'mozilla-central',
+      'help': "Branches to query results from",
+      }],
+    [['-b', '--build-type'],
+     {'default': 'opt',
+      'help': "Build type (default: opt)",
+      }],
+    [['--from'],
+     {'dest': 'from_date',
+      'default': 'today-week',
+      'help': "Starting date to pull data from, defaults "
+              "to a week ago",
+      }],
+    [['--to'],
+     {'dest': 'to_date',
+      'default': 'eod',  # end of day
+      'help': "Ending date to pull data from, defaults "
+              "to now",
+      }],
+    [['--path'],
+     {'required': True,
+      'default': 'dom/indexedDB',
+      'help': "Path relative to repository root (file or directory)",
+      }],
+    [['-p', '--platform'],
+     {'default': 'windows10-64',
+      'help': "Platform to limit results to (default: windows10-64)",
+      }],
+    [['-r', '--revision'],
+     {'dest': 'rev',
+      'required': True,
+      'default': '5b33b070378a',
+      'help': "Revision to limit results to",
+      }],
+    [['-t', '--test'],
+     {'required': True,
+      'dest': 'test_name',
+      'help': "Path to a test file",
+      }],
+
+    [['--limit'],
+     {'type': int,
+      'default': 10,
+      'help': "limit the number of rows in result"
+      }],
+
+    [['--format'],
+     {'type': str,
+      'default': 'table',
+      'help': "format of result"
+      }],
+    [['--sort-key'],
+     {'type': int,
+      'dest': 'sort_key',
+      'default': 4,
+      'help': "Key to sort on (int, 0-based index)",
+      }]
+]
+"""
+These are commonly used arguments which can be re-used. They are shared to
+provide a consistent CLI across recipes/queries.
+"""
+
+
+class ZeroContextEvaluator(ExpressionEvaluator):
```

Comment:
I should have guessed you would go ahead and implement this :+1: 

Extracting tokens out of the query files is a good case for a unit test, please add a test along with this PR.

Reply:
Thank you, I made the test. Please refer to https://github.com/mozilla/active-data-recipes/pull/85